Import Required Libraries

In [2]:
import pandas as pd
from sklearn.preprocessing import LabelEncoder
from sklearn.model_selection import train_test_split
from sklearn.svm import SVC
from sklearn.metrics import accuracy_score, classification_report

pandas: For loading and handling the dataset.

LabelEncoder: To convert categorical values to numeric format.

train_test_split: To divide the dataset into training and testing sets.

SVC: Support Vector Classifier from sklearn.

accuracy_score, classification_report: To evaluate the model.

Load the Dataset

In [3]:
df=pd.read_csv("car_evaluation.csv")
df

vhigh vhigh.1      2   2.1  small   low  unacc
0     vhigh   vhigh      2     2  small   med  unacc
1     vhigh   vhigh      2     2  small  high  unacc
2     vhigh   vhigh      2     2    med   low  unacc
3     vhigh   vhigh      2     2    med   med  unacc
4     vhigh   vhigh      2     2    med  high  unacc
...     ...     ...    ...   ...    ...   ...    ...
1722    low     low  5more  more    med   med   good
1723    low     low  5more  more    med  high  vgood
1724    low     low  5more  more    big   low  unacc
1725    low     low  5more  more    big   med   good
1726    low     low  5more  more    big  high  vgood

[1727 rows x 7 columns]

Rename Columns

In [4]:
df.columns = ['buying', 'maint', 'doors', 'persons', 'lug_boot', 'safety', 'class']


Sometimes column names are not properly loaded or have .1, .2, etc.

We rename them to meaningful names.

Encode Categorical Data

In [5]:
le = LabelEncoder()
for col in df.columns:
    df[col] = le.fit_transform(df[col])


All columns are categorical (e.g., low, med, high), so we need to convert them to numbers.

LabelEncoder replaces each category with a number.

Example: low → 0, med → 1, high → 2

Split Features and Target

In [6]:
X = df.drop('class', axis=1)
y = df['class']


X contains all input features (columns except class).

y contains the target/output class (which we want to predict).

Split the Data into Train and Test Sets



In [7]:
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, random_state=42)


train_test_split divides the data:

80% → Training

20% → Testing



Train the SVM Model

In [8]:
svm_model = SVC(kernel='rbf')
svm_model.fit(X_train, y_train)


SVC()

SVC is used to create the model.

kernel='rbf' is the Radial Basis Function kernel (default, works well in most cases).

.fit() trains the model using the training data.



Make Predictions

In [9]:
y_pred = svm_model.predict(X_test)

Use the trained model to predict the class labels for test data.

 Evaluate the Model

In [10]:
print("Accuracy:", accuracy_score(y_test, y_pred))
print("\nClassification Report:\n", classification_report(y_test, y_pred))


Accuracy: 0.8988439306358381

Classification Report:
               precision    recall  f1-score   support

           0       0.78      0.79      0.79        77
           1       1.00      0.13      0.24        15
           2       0.94      1.00      0.97       237
           3       0.86      0.71      0.77        17

    accuracy                           0.90       346
   macro avg       0.89      0.66      0.69       346
weighted avg       0.90      0.90      0.88       346



The model correctly predicted ~89.9% of the test data.

This is a good accuracy, indicating the model performs well overall.



Class 2 (good):

Best performance: Precision = 0.94, Recall = 1.00 → almost perfect predictions.

Model predicts 'good' cars very accurately.

Class 0 (unacc):

Fair performance (Precision = 0.78, Recall = 0.79).

Class 3 (vgood):

Decent prediction quality (Precision = 0.86, Recall = 0.71).

Class 1 (acc):

Poor recall (0.13): Model is missing many 'acc' class samples.

It detects very few actual 'acc' cars correctly, though precision is 1.00 (when it does predict ‘acc’, it’s correct—but it rarely predicts it).

